src/utils/canvasUtils.ts

```diff
@@ -68,26 +105,77 @@ export const drawVerticalLine = ({
   height,
   lineWidth,
   color,
-  lineCap
-}: any) => {
-  ctx.globalAlpha = 1;
+  lineCap = '',
+  opacity = 1,
+  dashSegments = []
+}: VerticalLineInterface) => {
+  ctx.globalAlpha = opacity;
   ctx.strokeStyle = color;
   ctx.lineWidth = lineWidth;
   ctx.lineCap = lineCap;
   ctx.beginPath();
+  ctx.setLineDash(dashSegments);
+
   ctx.moveTo(x, y);
   ctx.lineTo(x, height);
   ctx.stroke();
 };
 
+export const drawArea = ({
+  ctx,
+  points,
+  lineWidth,
+  lineCap = '',
+  opacity = 1,
+  dashSegments = [],
+  fill
+}: AreaInterface) => {
+  ctx.globalAlpha = opacity;
+  ctx.lineWidth = lineWidth;
+  ctx.lineCap = lineCap;
+  ctx.beginPath();
+  ctx.setLineDash(dashSegments);
+  ctx.moveTo(points[0].x, points[0].y);
+
+  points.forEach((p: {x: number; y: number; color?: string}, i: number) => {
```

Comment:
p => point
And please create separate type or interface for this. It's hard to read this line